### PR TITLE
[Ceph-Mgr] [Tier-2] Validate ceph insights

### PR DESCRIPTION
**Pull request by pranavprakash20**
# Description

Tests included :
CEPH-83573688    Ceph Insights : Delete insights reports within an interval
CEPH-83573810    Ceph Insights : Monitor log growth to ceph.audit.log during a failure scenario

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pranavprakash20** [comment]
Run log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-QGBFF4/

**Comment by pranavprakash20** [comment]
Adding DNM as second scenario needs further clarifications

File: cli/ceph/ceph.py [hunk 1]
```diff
@@ -43,9 +43,13 @@ def fsid(self):
             return out[0].strip()
         return out
 
-    def insights(self):
+    def insights(self, prune=False, hours=None):
         """Returns ceph insights"""
```
Comment (vamahaja) [inline cli/ceph/ceph.py:47]: Add doc string for method

File: tests/mgr/test_cephmgr_validate_insights.py [hunk 2]
```diff
@@ -54,6 +59,26 @@ def _ceph_run_insight(node, step_to_validate):
                 raise CephInsightOpsFailure(
                     f"Insights data doesn't have {section} section"
                 )
+    if step_to_validate == "verify-prune-health":
+        # Verify whether the prune-health command deletes the report generated
+        prune_timeout = 0  # Delete all logs
+        out = CephAdm(node).ceph.insights(prune=True, hours=prune_timeout)
+        sleep(prune_timeout)
```
Comment (vamahaja) [inline tests/mgr/test_cephmgr_validate_insights.py:66]: Constant sleep ? Try to check for some events to verify completion of previous command.

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Comment by pranavprakash20** [comment]
Updated run log:  http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-GGU76J/

File: tests/mgr/test_cephmgr_validate_insights.py [hunk 3]
```diff
@@ -54,6 +58,40 @@ def _ceph_run_insight(node, step_to_validate):
...
+        if out:
+            CephInsightOpsFailure("Failed to get ceph insights report after prune")
+        new_insights_data = json.loads(out)
+        if new_insights_data == insight_data:
+            CephInsightOpsFailure("The insights report has not been cleared")
+
+    if step_to_validate == "verify-log":
+        # Verify the ceph.audit.log has health history recorded and is not consuming more space
+        # and Excessive logging is not found
+        logs, _ = node.exec_command(cmd="cat /var/log/ceph/cephadm.log")
```
Comment (vamahaja) [inline tests/mgr/test_cephmgr_validate_insights.py:79]: this log file could be a huge. instead of cat, download to temp location and read.

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Comment by pranavprakash20** [comment]
Updated run log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-7YTHZ8/

File: ceph/ceph.py [hunk 4]
```diff
@@ -2096,6 +2096,17 @@ def upload_file(self, src, dst, sudo=False):
         client = self.rssh if sudo else self.ssh
         client().open_sftp().put(src, dst)
 
+    def download_file(self, src, dst, sudo=False):
```
Comment (vamahaja) [inline ceph/ceph.py:2099]: Call same method for CephObject as well. Refer - https://github.com/red-hat-storage/cephci/blob/master/ceph/ceph.py#L2153

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: ceph/ceph.py (hunk 4 above)
Comment (pranavprakash20) [reply ceph/ceph.py:2099]: Added

**Review by vamahaja — COMMENTED** [review]
LGTM

**Review by MohitBis — APPROVED** [review]
LGTM

**Review by vamahaja — APPROVED** [review]

**Pull request merged**